build-logic/src/main/kotlin/org.jabref.gradle.base.dependency-rules.gradle.kts

```diff
@@ -293,52 +293,38 @@ extraJavaModuleInfo {
     module("com.github.javaparser:javaparser-symbol-solver-core", "com.github.javaparser.symbolsolver.core")
     module("net.sf.jopt-simple:jopt-simple", "jopt.simple")
 
-    // "com.github.eclipse:org.eclipse.lsp4j", "lsp4j"
-    //   - The name 'org.eclipse.lsp4j' is different than the name derived from the Jar file name 'lsp4j'; turn off 'failOnModifiedDerivedModuleNames' or explicitly allow override via 'overrideModuleName()'
-    //   - Not a module and no mapping defined: lsp4j-0.24.0.jar
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j", "lsp4j") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j", "org.eclipse.lsp4j") {
         exportAllPackages()
         requireAllDefinedDependencies()
-        // Note the missing "lsp4j" at the group
-        mergeJar("com.github.eclipse:lsp4j")
         requires("com.google.gson")
-
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.debug", "lsp4j.debug") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.debug", "org.eclipse.lsp4j.debug") {
         exportAllPackages()
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.generator", "lsp4j.generator") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.generator", "org.eclipse.lsp4j.generator") {
         exportAllPackages()
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc", "lsp4j.jsonrpc") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc", "org.eclipse.lsp4j.jsonrpc") {
         exportAllPackages()
         requires("com.google.gson")
         requires("java.logging")
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc.debug", "lsp4j.jsonrpc.debug") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.jsonrpc.debug", "org.eclipse.lsp4j.jsonrpc.debug") {
         exportAllPackages()
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.websocket", "lsp4j.websocket") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.websocket", "org.eclipse.lsp4j.websocket") {
         exportAllPackages()
         requireAllDefinedDependencies()
     }
-    module("com.github.eclipse.lsp4j:org.eclipse.lsp4j.websocket.jakarta", "lsp4j.websocket.jakarta") {
-        overrideModuleName()
+    module("org.eclipse.lsp4j:org.eclipse.lsp4j.websocket.jakarta", "org.eclipse.lsp4j.websocket.jakarta") {
         exportAllPackages()
         requireAllDefinedDependencies()
     }
     module("jakarta.websocket:jakarta.websocket-api", "jakarta.websocket") {
         overrideModuleName()
         exportAllPackages()
     }
-    module("javax.websocket:javax.websocket-api", "javax.websocket") {
-        overrideModuleName()
+    module("javax.websocket:javax.websocket-api", "javax.websocket.api") {
```

Comment:
Should this adjustment be applied to `module("jakarta.websocket:jakarta.websocket-api", "jakarta.websocket")` as well?

Reply:
Oh, sorry my mistake. It's alright. Jakarta does explicitly declare the module name
```
module jakarta.websocket {
    exports jakarta.websocket.server;
    
    requires transitive jakarta.websocket.client;
    
    uses jakarta.websocket.server.ServerEndpointConfig.Configurator;
}
```
For javax.websocket, it's a different story since its automatic module name is indeed `javax.websocket.api`. Sorry for the wrong alert!